### DISCUSSION TITLE
[DONE] Change the Maven Project Artifact Name and Version Number

### DISCUSSION THREAD

**Discussion by bglamadrid** [trebol-ecommerce/trebol-backend-monolith#228] [Announcements]
You may have read that the main focus behind Trébol is "to be highly adaptive and flexible". To make those things possible, its inner workings must be **well-tested and easy to navigate**. Things should be easy to understand, and therefore easy to "hack". There is still work to be done in both parts.

Trébol so far hasn't been a production-ready software. I didn't take this in account when I set a `v1.0.0` mark and onwards.

I want to rectify that.

---

So, here's what I'll do: "start over" by setting:

- a new Maven artifact (combination of `groupId`/`artifactId` in the `pom.xml`)
- version number back to `0.1.0`

The point is that this artifact should reach a `v1.0.0` when better conditions are met.
I, for one, think it should at least include more features (exposed through RESTful APIs with all due security measures and so), at least to make its users able to:

- edit most application parameters, temporarily and/or permanently
- manage the sets of permissions that are made available to each user role

I hope this annoucement clears any confusion, if you ever happen to notice this strange change in versions.

**Comment by NyorJa** [comment]
Hi @bglamadrid when will be the version will be incremented? What is our strategy to this?

**Reply by bglamadrid** [reply]
Since this project isn't being published to any public Maven artifact repository, this doesn't really break anything. It just might confuse those unaware.

For this I'll just follow the usual flow, creating a branch for this precise purpose, then create its PR and eventually merge it to `main`.

If you do have changes that you want to merge however, that's another thing entirely.

**Comment by bglamadrid** [comment]
If anyone has been working from an older `main` base branch since at least two months ago, and is now met with these latest changes, don't hesitate to reach out. I'll gladly help to resolve conflicts.

**Comment by bglamadrid** [comment]
It is now done, this repo name as well as the artifact and version number are updated.
